### ISSUE TITLE
Configurable Working Directory for Exported Credential

### ISSUE THREAD

**Issue by ccjernigan** [google-github-actions/auth#134]
### TL;DR

I'd like to put the credential automatically generated by this action into a different directory from `GITHUB_WORKSPACE`.

### Detailed design

```markdown
steps:
    - name: Authenticate to Google Cloud
      uses: google-github-actions/auth@v0.6.0
      with:
        project_id: ${{ inputs.google-cloud-project }}
        service_account: ${{ inputs.google-cloud-service-account }}
        workload_identity_provider: ${{ inputs.google-cloud-workload-identity-provider }}
        working_directory: ${{ env.home }} <-- Possible new parameter to change working directory of the action.
```

### Additional information

Although the credentials are short-lived, placing them in `GITHUB_WORKSPACE` creates some additional opportunities to accidentally expose the credential for a period of time.

I managed to do this accidentally a few days ago.  Here's what happened:
1. Ran the auth step to generate a token
2. Ran a step to run tests on Firebase test lab
3. Ran a step to upload the test results using a bash script with a `zip` command.  Due to a typo in the `zip` command, it compressed the entire `GITHUB_WORKSPACE` directory instead of just the output files I wanted it to collect.

Placing the secrets in a different directory, like `HOME` would have avoided this issue.  GitHub Actions steps use a default working directory of `GITHUB_WORKSPACE`, so including the secret in that path makes it more likely that downstream steps will accidentally leak it.

I don't have specific requirements on the exact implementation.  An optional parameter for the working directory would been  reasonable option.  This would keep backwards compatibility with existing users, while granting flexibility for where the secret is written.

**Comment by sethvargo** [comment]
As of 0.6.0, the credentials file follows a well-known naming format:

```text
gha-creds-*.json
```

As noted in the [Prerequisites](https://github.com/google-github-actions/auth#prerequisites) and the [Troubleshooting Guide](https://github.com/google-github-actions/auth/blob/main/docs/TROUBLESHOOTING.md#dirty-git-or-bundled-credentials), we recommend adding this pattern to your `.gitignore`.

We've been down the path of allowing users to specify the output path in setup-gcloud, and it turns out to be a bad idea. There's an _even higher_ chance of the credential being accidentally committed into a binary or container, and it introduces many challenges with self-hosted runners, docker-based actions, and in post-job cleanup.

You can use `git` to create an archive, which will take into account your .gitignore:

```shell
git archive -o file.zip HEAD
```

I personally have this as a shell function:

```shell
gitzip() {
  local NAME="${1:-arhive}"
  git archive -o "${NAME}.zip" HEAD
}
```

Please let me know if you have additional questions. Thanks!

**Comment by ccjernigan** [comment]
How does adding the key to the gitignore help?

We're trying to collect specific build outputs, which are also ignored by git.  For example in our repo containing a Gradle project, all `build` folders are ignored.  Unless I missed something, it seems that trying to leverage the gitignore means we won't collect anything of interest.

**Comment by sethvargo** [comment]
Why don't you compile your build artifacts into a subfolder then? That way you don't have to worry about any pollution. The same would be true if you ran any tests that generated files in the working directory.

We've been down the path of a customizable working directory, and it introduces far too many security risks.
